packages/server/app/lib/auth.ts

```diff
@@ -1,48 +1,80 @@
-import { createSessionStorage } from "./session";
+import bcrypt from "bcryptjs";
+import jwt from "jsonwebtoken";
+import { createJWTCookie, clearJWTCookie } from "./session";
+import { User } from "./types";
 
-export async function login(request: Request, password: string, env: Env) {
-  if (password !== env.CF_APP_PASSWORD) {
+export async function login(_request: Request, password: string, env: Env) {
+  const isValidPassword = await bcrypt.compare(password, env.CF_PASSWORD_HASH);
+  if (!isValidPassword) {
     throw new Error("Invalid password");
   }
 
-  const sessionStorage = createSessionStorage(env.CF_APP_PASSWORD);
-  const session = await sessionStorage.getSession(request.headers.get("Cookie"));
-  
-  session.set("authenticated", true);
+  const token = jwt.sign(
+    { 
+      authenticated: true,
+      iat: Math.floor(Date.now() / 1000),
+    },
+    env.CF_JWT_SECRET,
+    { 
+      expiresIn: '30d',
+      issuer: 'counterscale'
+    }
+  );
   
   return redirect("/dashboard", {
     headers: {
-      "Set-Cookie": await sessionStorage.commitSession(session),
+      "Set-Cookie": createJWTCookie(token),
     },
   });
 }
 
-export async function logout(request: Request, env: Env) {
-  const sessionStorage = createSessionStorage(env.CF_APP_PASSWORD);
-  const session = await sessionStorage.getSession(request.headers.get("Cookie"));
-  
+export async function logout(_request: Request, _env: Env) {
   return redirect("/", {
     headers: {
-      "Set-Cookie": await sessionStorage.destroySession(session),
+      "Set-Cookie": clearJWTCookie(),
     },
   });
 }
 
 export async function requireAuth(request: Request, env: Env) {
-  const sessionStorage = createSessionStorage(env.CF_APP_PASSWORD);
-  const session = await sessionStorage.getSession(request.headers.get("Cookie"));
+  const user = await getUser(request, env);
   
-  if (!session.get("authenticated")) {
+  if (!user.authenticated) {
     throw redirect("/");
   }
   
-  return session;
+  return user;
 }
 
-export async function getUser(request: Request, env: Env) {
-  const sessionStorage = createSessionStorage(env.CF_APP_PASSWORD);
-  const session = await sessionStorage.getSession(request.headers.get("Cookie"));
-  
-  return session.get("authenticated") ? { authenticated: true } : null;
+export async function getUser(request: Request, env: Env): Promise<User> {
+  try {
+    const cookieHeader = request.headers.get("Cookie");
+    if (!cookieHeader) {
+      return { authenticated: false };
+    }
+
+    // Extract JWT from cookie
+    const cookies = cookieHeader.split(';').reduce((acc, cookie) => {
+      const [key, value] = cookie.trim().split('=');
+      acc[key] = value;
+      return acc;
+    }, {} as Record<string, string>);
+
+    const token = cookies['__counterscale_token'];
+    if (!token) {
+      return { authenticated: false };
+    }
+
+    const decoded = jwt.verify(token, env.CF_JWT_SECRET) as jwt.JwtPayload;
```

Comment:
Do you have to pass `maxAge` or something here? How do we ensure the token is still valid (< 30 days)?

Reply:
it's already encoded in the token when it gets signed!